### PR TITLE
(BIDS-2461) fix response codes

### PR DESCRIPTION
**Pull request by LuccaBitfly**
### <samp>🤖 Generated by Copilot at f19a4da</samp>

This pull request fixes the HTTP status codes for various endpoints in the `handlers` package to use 500 (Internal Server Error) instead of 503 (Service Unavailable) when there are server-side errors. This improves the consistency and accuracy of the error handling and communication with the API clients. It also removes some unused or commented out code from the `cmd/explorer/main.go` and `handlers/confirmation.go` files.

### CODE REVIEW

File: handlers/api.go [hunk 1]
```diff
@@ -3048,7 +3048,7 @@ func RegisterEthpoolSubscription(w http.ResponseWriter, r *http.Request) {
 
 	subscriptionCount, err := db.GetAppSubscriptionCount(claims.UserID)
 	if err != nil || subscriptionCount >= 5 {
-		w.WriteHeader(http.StatusInternalServerError)
+		w.WriteHeader(http.StatusConflict)
 		sendErrorResponse(w, r.URL.String(), "reached max subscription limit")
```
Comment (Eisei24) [inline handlers/api.go:3052]: **issue:** This combination doesn't seem to make sense.
First we set the header to **StatusConflict** even if we have a normal error.
Then we set the header to **StatusBadRequest** in `sendErrorResponse`.

**Review by Eisei24 — CHANGES_REQUESTED** [review]
Please have a look at my comments.

File: handlers/advertisewithus.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ func AdvertiseWithUs(w http.ResponseWriter, r *http.Request) {
 	pageData.FlashMessage, err = utils.GetFlash(w, r, "ad_flash")
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/advertisewithus.go:26]: IMO this should be looked at in its own issue and would go beyond the scope of this PR.

File: handlers/broadcast.go [hunk 3]
```diff
@@ -98,7 +98,7 @@ func BroadcastStatus(w http.ResponseWriter, r *http.Request) {
 		if err != sql.ErrNoRows {
 			logger.Errorf("error retrieving job %v", err)
 		}
-		http.Error(w, "Internal server error", http.StatusServiceUnavailable)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/broadcast.go:101]: fixed, also maybe worthy of creating a BIDS for this:
image

File: handlers/api.go [hunk 4]
```diff
@@ -104,7 +104,7 @@ func ApiHealthz(w http.ResponseWriter, r *http.Request) {
 
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
-			http.Error(w, "No monitoring data available", http.StatusServiceUnavailable)
+			http.Error(w, "No monitoring data available", http.StatusNotFound)
 			return
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/api.go:108]: fixed

File: handlers/eth1Account.go [hunk 5]
```diff
@@ -52,7 +52,7 @@ func Eth1Address(w http.ResponseWriter, r *http.Request) {
 	metadata, err := db.BigtableClient.GetMetadataForAddress(addressBytes, 0, db.ECR20TokensPerAddressLimit)
 	if err != nil {
 		logger.Errorf("error retieving balances for %v route: %v", r.URL.String(), err)
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/eth1Account.go:54]: fixed

File: handlers/stripe.go [hunk 6]
```diff
@@ -260,41 +259,28 @@ func StripeWebhook(w http.ResponseWriter, r *http.Request) {
...
 			logger.WithError(err).Error("error creating subscription no items found", subscription)
-			http.Error(w, "error creating subscription no items found", http.StatusServiceUnavailable)
+			http.Error(w, "error creating subscription no items found", http.StatusInternalServerError)
 			return
 		}
 
 		if len(subscription.Items.Data) == 0 {
 			logger.WithError(err).Error("error creating subscription no items found", subscription)
-			http.Error(w, "error creating subscription no items found", http.StatusServiceUnavailable)
+			http.Error(w, "error creating subscription no items found", http.StatusInternalServerError)
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/stripe.go:283]: Should also be bad request, fixed

File: handlers/stripe.go [hunk 7]
```diff
@@ -314,19 +300,19 @@ func StripeWebhook(w http.ResponseWriter, r *http.Request) {
...
 
 		if subscription.Items == nil {
 			utils.LogError(nil, fmt.Errorf("error updating subscription no items found %v", subscription), 0)
-			http.Error(w, "error updating subscription no items found", http.StatusServiceUnavailable)
+			http.Error(w, "error updating subscription no items found", http.StatusBadRequest)
 			return
 		}
 
 		if len(subscription.Items.Data) == 0 {
 			utils.LogError(nil, fmt.Errorf("error updating subscription no data found %v", subscription), 0)
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/stripe.go:314]: This is not user input, this is a private endpoint used by Stripe, in which case erroneous requests warrant error logs imo.

File: handlers/stripe.go [hunk 8]
```diff
@@ -428,47 +414,44 @@ func StripeWebhook(w http.ResponseWriter, r *http.Request) {
...
-			http.Error(w, "error creating transaction :"+err.Error(), http.StatusServiceUnavailable)
+			http.Error(w, "error creating transaction :"+err.Error(), http.StatusInternalServerError)
 		}
 		defer tx.Rollback()
 
 		err = db.StripeUpdateSubscriptionStatus(tx, invoice.Lines.Data[0].Subscription, true, nil)
 		if err != nil {
 			logger.WithError(err).Error("error processing invoice failed to activate subscription for customer", invoice.Customer.ID)
-			http.Error(w, "error proccesing invoice failed to activate subscription for customer", http.StatusServiceUnavailable)
+			http.Error(w, "error proccesing invoice failed to activate subscription for customer", http.StatusInternalServerError)
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/stripe.go:446]: IMO double c is the superior spelling of this word, but fixed

File: handlers/monitoring.go [hunk 9]
```diff
@@ -41,7 +41,7 @@ func Monitoring(w http.ResponseWriter, r *http.Request) {
 
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
-			http.Error(w, "No monitoring data available", http.StatusServiceUnavailable)
+			http.Error(w, "No monitoring data available", http.StatusNotFound)
 			return
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/monitoring.go:45]: fixed

File: handlers/validatorRewards.go [hunk 10]
```diff
@@ -220,7 +220,7 @@ func RewardNotificationSubscribe(w http.ResponseWriter, r *http.Request) {
 
 	if err != nil || count >= 5 {
 		logger.WithField("route", r.URL.String()).Info(fmt.Sprintf("User Subscription limit (%v) reached %v", count, err))
-		http.Error(w, "Internal server error, User Subscription limit reached", http.StatusInternalServerError)
+		http.Error(w, "Internal server error, User Subscription limit reached", http.StatusConflict)
 		return
 	}
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/validatorRewards.go:225]: nice catch! the error message was deceiving, fixed

File: handlers/validatorRewards.go [hunk 11]
```diff
@@ -239,7 +239,7 @@ func RewardNotificationSubscribe(w http.ResponseWriter, r *http.Request) {
 
 	if validatorArr == "" || !isValidCurrency(currency) {
 		logger.WithField("route", r.URL.String()).Error("Bad Query")
-		http.Error(w, "Internal server error, Bad Query", http.StatusInternalServerError)
+		http.Error(w, "Internal server error, Bad Query", http.StatusBadRequest)
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/validatorRewards.go:242]: fixed

File: handlers/validatorRewards.go [hunk 12]
```diff
@@ -239,7 +239,7 @@ func RewardNotificationSubscribe(w http.ResponseWriter, r *http.Request) {
 
 	if validatorArr == "" || !isValidCurrency(currency) {
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/validatorRewards.go:240]: fixed

File: handlers/validatorRewards.go [hunk 13]
```diff
@@ -283,7 +283,7 @@ func RewardNotificationUnsubscribe(w http.ResponseWriter, r *http.Request) {
...
 		logger.WithField("route", r.URL.String()).Error("Bad Query")
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/validatorRewards.go:285]: fixed

File: handlers/user.go [hunk 14]
```diff
@@ -944,7 +771,7 @@ func UserNotificationsData(w http.ResponseWriter, r *http.Request) {
 	balances, err := db.BigtableClient.GetValidatorBalanceHistory(indices, services.LatestEpoch(), services.LatestEpoch())
 	if err != nil {
 		logger.WithError(err).WithField("route", r.URL.String()).Errorf("error retrieving validator balance data")
-		http.Error(w, "Internal server error", http.StatusServiceUnavailable)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
```
[the remark this answers is not in the document]
Comment (LuccaBitfly) [reply handlers/user.go:774]: fixed

File: handlers/api.go (hunk 1 above)
Comment (Eisei24) [reply handlers/api.go]: **question:** The new changes alter the way we fill `w`.
Before it was a
```
j := json.NewEncoder(w)
response := &types.ApiResponse{}
response.Status = errMsg
j.Encode(response)
```
now it is a
`fmt.Fprintln(w, errorMsg)`

Is such a change allowed?

**Review by Eisei24 — COMMENTED** [review]
Please have a look at the last remaining question.

File: handlers/api.go (hunk 1 above)
Comment (LuccaBitfly) [reply handlers/api.go]: even nicer catch, fixed

Comment (Eisei24) [reply handlers/api.go]: It was fixed for one of two cases, please also fix the other, just search for USER_SUBSCRIPTION_LIMIT in api.go.

**Review by Eisei24 — CHANGES_REQUESTED** [review]
Please fix the other API error case.

**Review by Eisei24 — APPROVED** [review]
lgtm

**Pull request merged**